pkg/services/updatemanager/plugins_test.go

```diff
@@ -300,3 +304,88 @@ func (c *fakeHTTPClient) Do(req *http.Request) (*http.Response, error) {
 
 	return resp, nil
 }
+
+func TestPluginsService_PluginsAutoUpdateFlag(t *testing.T) {
+	updateCheckURL, _ := url.Parse("https://grafana.com/api/plugins/versioncheck")
+
+	tests := []struct {
+		name         string
+		flagEnabled  bool
+		expectUpdate bool
+	}{
+		{
+			name:         "pluginsAutoUpdate enabled calls updateAll",
+			flagEnabled:  true,
+			expectUpdate: true,
+		},
+		{
+			name:         "pluginsAutoUpdate disabled does not call updateAll",
+			flagEnabled:  false,
+			expectUpdate: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			setupOpenFeatureProvider(t, tt.flagEnabled)
+
+			updateCallCount := 0
+
+			availableUpdates := map[string]availableUpdate{
+				"test-plugin": {
+					localVersion:     "0.9.0",
+					availableVersion: "1.0.0",
+				},
+			}
+
+			svc := &PluginsService{
+				availableUpdates: availableUpdates,
+				httpClient: &fakeHTTPClient{
+					fakeResp: `[]`,
+				},
+				log:            log.NewNopLogger(),
+				tracer:         tracing.InitializeTracerForTest(),
+				updateCheckURL: updateCheckURL,
+				updateChecker:  pluginchecker.ProvideService(managedplugins.NewNoop(), provisionedplugins.NewNoop(), &mockPluginPreinstall{}),
+				pluginStore:    &pluginstore.FakePluginStore{PluginList: []pluginstore.Plugin{}},
+				pluginInstaller: &pluginfakes.FakePluginInstaller{
+					AddFunc: func(ctx context.Context, pluginID, version string, opts plugins.AddOpts) error {
+						updateCallCount++
+						return nil
+					},
+				},
+				grafanaVersion: "10.0.0",
+			}
+
+			ctx := context.Background()
+			// Test the synchronous initialization work directly, without the long-running ticker loop
+			svc.checkAndUpdate(ctx)
+
+			if tt.expectUpdate {
+				require.Greater(t, updateCallCount, 0, "updateAll should be called when flag is enabled")
```

Comment:
We could probably be more explicit here:
```suggestion
				require.Equal(t, updateCallCount, 1, "updateAll should be called when flag is enabled")
```